BookingProject/BookingWebAPI/BookingWebApi.Application/Services/AuthenticationService.cs

```diff
@@ -0,0 +1,124 @@
+﻿using BookingWebApi.Application.Decorators;
+using BookingWebApi.Application.DTOs.AppUserDTOs;
+using BookingWebApi.Application.Interfaces;
+using BookingWebApi.Application.Models;
+using BookingWebApi.Domain.Entities;
+using FluentValidation;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BookingWebApi.Application.Services
+{
+    public class AuthenticationService : IAuthenticationService
+    {
+        private readonly IUserManagerDecorator<AppUser> _userManager;
+        private readonly ITokenService _tokenService;
+        private readonly ISignInManagerDecorator<AppUser> _signInManager;
+        private readonly IValidator<RegisterDto> _validatorRegisterDto;
+        private readonly IValidator<LoginDto> _validatoLoginDto;
```

Comment:
I dislike this approach of validation - by presenting other models which require validation, you will need to inject more and more validators.

I propose to make a custom validation middleware, where you will use Reflection to get validator for the model.
This way your class will have less dependencies and you extract validation logic to a separate place.

Described fix is described in this video: https://www.youtube.com/watch?v=OhQA4PDdp0Q but instead of ValidationBehavior you will have validation middleware.

If this looks too complex at this moment, you can leave it as is for now